jena-fuseki2/jena-fuseki-main/src/main/java/org/apache/jena/fuseki/main/FusekiServer.java

```diff
@@ -501,6 +505,18 @@ public Builder enableMetrics(boolean withMetrics) {
             return this;
         }
 
+        /** Add the "/$/compact/*" servlet that triggers compaction for specified dataset. */
+        public Builder enableCompact(boolean withCompact) {
+            this.withCompact = withCompact;
+            return this;
+        }
+
+        /** Add the "/$/tasks" servlet that responds with info about tasks run on the server */
+        public Builder enableTasks(boolean withTasks) {
```

Comment:
Does it not make more sense to just automatically enable tasks if any background task feature is enabled?

Reply:
I considered this. But just because you're enabling a background task, that doesn't always mean that someone might want/need to track those tasks, so I decided to make it opt-in. But if automatically including it is what is wanted, then I can add it.

Reply:
Yeah I can see arguments for both approaches.

We recently had one user who was trying to `cron` a script that was requesting a compaction and then immediately deleting files out from under Fuseki/TDB ultimately causing it to crash.  Our response to them was to refer them to using the tasks API to only clean up files after compaction was finished.

So my strong preference is always to have task monitoring enabled if we have any of the potentially long running background task APIs enabled.

Be interested to see what @kinow and @afs's opinions are as they use Fuseki more heavily than me

Reply:
I am not maintaining a production Jena instance at the moment. But at least for me I prefer to have the tasks endpoint accessible to that I can either check task statuses manually via command line `curl`, or preferably via Fuseki UI.

Reply:
sounds like there is a consensus to add it by default. I guess if someone wants to have it disabled, an opt-out option could always be added in the future.

Reply:
OK, I've updated the code to enable tasks by default if compact is enabled.